ignition: enable systemd firstboot condition through kargs

runcom opened the pull request:
This is a workaround to make the systemd believe it's firstboot when ignition runs on real firstboot.
Right now, since we ship /etc/machine-id, systemd thinks it's not firstboot and ignition depends on it to run on the real firstboot to enable services from presets. Since this only applies to artifacts with ignition and changing machineid-compat at commit creation time may have undesiderable effect, we're doing it here as a stopgap. We may revisit this in the future.

Requires https://github.com/osbuild/osbuild/pull/1249


This pull request includes:

- [ ] adequate testing for the new functionality or fixed issue
- [ ] adequate documentation informing people about the change such as
  - [ ] submit a PR for the [guides](https://github.com/osbuild/guides) repository if this PR changed any behavior described there: https://www.osbuild.org/guides/

<!--
Thanks for proposing a change to osbuild-composer!

Please don't remove the above check list. These are things that each pull
request must have before it is merged. It helps maintainers to not forget
anything.

If the reason for ticking any of the boxes is ambiguous, please add a short
note explaining why.

In addition, if this pull request fixes a downstream issue, please refer to
test/README.md and add these additional items:

- [ ] 1st commit of any `rhbz#` related PR contains bug reproducer; CI reports FAIL or
- [ ] PR contains automated tests for new functionality and
- [ ] QE has approved reproducer/new tests and
- [ ] Subsequent commits provide bug fixes without modifying the reproducer; CI reports PASS and
- [ ] QE approves this PR; RHBZ status is set to `MODIFIED + Verified=Tested`

Information regarding our GitLab pipeline (Schutzbot):

CI will not be ran automatically if WIP label is applied or the PR is in DRAFT state, instead only
a link will be provided to the pipeline which can then be triggered manually if desired. To run the
CI automatically either switch the PR to ready or apply WIP+test label.

Outside contributors need manual approval from one of the osbuild-composer maintainers.

Schutzbot will only be triggered if all Tests jobs in GitHub workflow succeed.
-->

thozza commented on internal/osbuild/ignition_stage.go:
```diff
@@ -6,6 +6,7 @@ import (
 )
 
 type IgnitionStageOptions struct {
+	Network []string `json:"network"`
```
This should be marked as `omitempty`, since this is not a required property.

thozza reviewed (commented):
This looks good. I expect that you will also need to regenerate image test manifests that use the commit-deployment pipeline.

thozza reviewed (approved):
LGTM 👍

atodorov commented on test/data/ansible/check_ostree.yaml:
```diff
@@ -71,6 +71,29 @@
+            that:
+              - "'Hello, World!' in result_hello_service_log.stdout"
+            fail_msg: "hello.service doesn't have the correct log"
+            success_msg: "hello.service started and working"
```
I don't understand how is this assertion related to `systemd.condition-first-boot=true` argument ? Can you clarify ?

atodorov replied:
FTR see https://redhat-internal.slack.com/archives/C0235DZB0DT/p1677592474917199

thozza commented:
Hi @runcom.

Please:

1. Squash the two commits into a single one or alternatively do the SPEC file and Schutzfile update in a single (optionally separate) commit.
2. Add a note about the reason to depend on osbuild v81 in the commit message that changes the dependency (basically note that it is because of the fixed Ignition stage).

Otherwise this looks good. Thanks 👍

runcom commented:
> * Squash the two commits into a single one or alternatively do the SPEC file and Schutzfile update in a single (optionally separate) commit.
> * Add a note about the reason to depend on osbuild v81 in the commit message that changes the dependency (basically note that it is because of the fixed Ignition stage).

squashed and added note, thanks @thozza

thozza commented:
All CI jobs passed, except for the VSphere ones, which are affected by a known DNS issue. Admin-merging, because this is an important bugfix.
